zebra-rpc/src/methods/tests/vectors.rs

```diff
@@ -30,4 +39,40 @@ async fn rpc_getinfo() {
     assert_eq!(get_info.subversion, USER_AGENT);
 
     mempool.expect_no_requests().await;
+    state.expect_no_requests().await;
+}
+
+#[tokio::test]
+async fn rpc_getblock() {
+    zebra_test::init();
+
+    // Number of blocks to populate state with
+    const NUMBER_OF_BLOCKS: u32 = 10;
+
+    // Put the first `NUMBER_OF_BLOCKS` blocks in a vector
+    let blocks: Vec<Arc<Block>> = zebra_test::vectors::MAINNET_BLOCKS
+        .range(0..=NUMBER_OF_BLOCKS)
+        .map(|(_, block_bytes)| block_bytes.zcash_deserialize_into::<Arc<Block>>().unwrap())
+        .collect();
+
+    let mempool: MockService<_, _, _, BoxError> = MockService::build().for_unit_tests();
+    // Create a populated state service
+    let state = zebra_state::populated_state(blocks.clone(), Network::Mainnet).await;
+
+    // Init RPC
+    let rpc = RpcImpl {
+        app_version: "Zebra version test".to_string(),
+        mempool: Buffer::new(mempool.clone(), 1),
+        state,
+    };
+
+    // Make calls and check response
+    for (i, block) in blocks.into_iter().enumerate() {
+        let get_block = rpc
+            .get_block(Height(i as u32), 0u8)
+            .await
+            .expect("We should have a GetBlock struct");
+
+        assert_eq!(get_block.data, block.into());
+    }
```

Comment:
Can we check there are no mempool requests here?
```suggestion
        mempool.expect_no_requests().await;
    }
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3707/commits/1d15df77d89d4dd0dd8bb584aea599af13aec791